xds/src/main/java/io/grpc/xds/client/ControlPlaneClient.java

```diff
@@ -167,9 +167,11 @@ void adjustResourceSubscription(XdsResourceType<?> resourceType) {
     resourceStore.startMissingResourceTimers(resources, resourceType);
 
     if (resources.isEmpty()) {
-      // The resource type no longer has subscribing resources; clean up references to it
+      // The resource type no longer has subscribing resources; clean up references to it.
+      // Except for nonces. If the resource type becomes used again the control plane can ignore
+      // requests for old/missing nonces. Old type's nonces are dropped when the ADS stream is
+      // restarted.
       versions.remove(resourceType);
```

Comment:
I think we should retain the version also.  The version doesn't change when we change the set of resources that we're subscribed to, and unsubscribing from all resources is just a special case of changing the set of resources that we're subscribed to.

Reply:
We had discussed this offline and Mark prefers keeping the version but doesn't see it as mandatory. We want to drop our resource data structures, and that means the version is also lost. The version serves no purpose when there are no resources saved by the client, so it can't make any difference which way it is done.

Reply:
I agree that it *probably* won't cause any problems to drop it.  But I do worry about potential edge cases we might hit in the future.

I guess let's kick the can down the road here and see if we have problems.